src/swagger/specGenerator.ts

```diff
@@ -6,6 +6,253 @@ import { Swagger } from './swagger';
+          description: referenceType.description,
+          properties: this.buildProperties(referenceType.properties),
+          required: required && required.length > 0 ? Array.from(new Set(required)) : undefined,
+          type: 'object',
+        };
+
+        if (referenceType.additionalProperties) {
+          definitions[referenceType.refName].additionalProperties = this.buildAdditionalProperties(referenceType.additionalProperties);
+        }
+
+        if (referenceType.example) {
+          definitions[referenceType.refName].example = referenceType.example;
+        }
+      }
+
+      // Enum definition
+      if (referenceType.enums) {
+        definitions[referenceType.refName] = {
+          description: referenceType.description,
+          enum: referenceType.enums,
+          type: 'string',
+        };
+      }
+    });
+
+    return definitions;
+  }
+
+  private buildPaths() {
+    const paths: { [pathName: string]: Swagger.Path } = {};
+
+    this.metadata.controllers.forEach(controller => {
+      const normalisedControllerPath = normalisePath(controller.path, '/');
+      // construct documentation using all methods except @Hidden
+      controller.methods.filter(method => !method.isHidden).forEach(method => {
+        const normalisedMethodPath = normalisePath(method.path, '/');
+        const path = normalisePath(`${normalisedControllerPath}${normalisedMethodPath}`, '/', '', false);
+        paths[path] = paths[path] || {};
+        this.buildMethod(controller.name, method, paths[path]);
+      });
+    });
+
+    return paths;
+  }
+
+  private buildMethod(controllerName: string, method: Tsoa.Method, pathObject: any) {
+    const pathMethod: Swagger.Operation = pathObject[method.method] = this.buildOperation(controllerName, method);
+    pathMethod.description = method.description;
+    pathMethod.summary = method.summary;
+    pathMethod.tags = method.tags;
+
+    // Use operationId tag otherwise fallback to generated. Warning: This doesn't check uniqueness.
+    pathMethod.operationId = method.operationId || pathMethod.operationId;
+
+    if (method.deprecated) {
+      pathMethod.deprecated = method.deprecated;
+    }
+
+    if (method.security) {
+      pathMethod.security = method.security as any[];
+    }
+
+    pathMethod.parameters = method.parameters
+      .filter(p => {
+        return !(p.in === 'request' || p.in === 'body-prop');
+      })
+      .map(p => this.buildParameter(p));
+
+    const bodyPropParameter = this.buildBodyPropParameter(controllerName, method);
+    if (bodyPropParameter) {
+      pathMethod.parameters.push(bodyPropParameter);
+    }
+    if (pathMethod.parameters.filter((p: Swagger.BaseParameter) => p.in === 'body').length > 1) {
+      throw new Error('Only one body parameter allowed per controller method.');
+    }
+
+    // Apply custom attributes
+    method.customAttributes.forEach((customAttr) => pathMethod[customAttr.key] = customAttr.value);
```

Comment:
This line should be the one moved, everything else added in this file should be removed.